src/librados/IoCtxImpl.cc

```diff
@@ -767,6 +767,22 @@ int librados::IoCtxImpl::aio_operate_read(const object_t &oid,
   return 0;
 }
 
+int librados::IoCtxImpl::cache_aio_operate_read(const object_t &oid, 
```

Comment:
I'm still suspicious of this being based on librados completion machinery, adding @adamemerson for comment

Reply:
Doesn't hurt exactly? If it's integrating on top of librados as is it makes sense to use the same completion machinery unless we have differing requirements like scatter-gather.

But it should definitely go on top of librados instead of being put into it, I think.